### PR TITLE
polygon/sync: download blocks

### PR DESCRIPTION
**Pull request by taratorio**

### CODE REVIEW

File: polygon/sync/bodies_verifier.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package sync
+
+import "github.com/ledgerwatch/erigon/core/types"
+
+type BodiesVerifier func(headers []*types.Header, bodies []*types.Body) error
+
+func VerifyBodies(_ []*types.Header, _ []*types.Body) error {
+	//
+	// TODO (subsequent PRs)
+	//
```
Comment (taratorio) [inline polygon/sync/bodies_verifier.go:10]: added to task list

File: polygon/sync/header_downloader.go [hunk 2]
```diff
@@ -54,25 +52,32 @@ func newHeaderDownloader(
...
 		headersVerifier:               headersVerifier,
-		headersWriter:                 headersWriter,
+		bodiesVerifier:                bodiesVerifier,
+		storage:                       storage,
 		notEnoughPeersBackOffDuration: notEnoughPeersBackOffDuration,
 	}
 }
 
+//
+// TODO (subsequent PRs) rename to block downloader? revise and update naming in logs, tests, etc.
```
Comment (taratorio) [inline polygon/sync/header_downloader.go:71]: https://github.com/ledgerwatch/erigon/pull/9681 - chained to this PR (first merge current PR)

File: polygon/sync/header_downloader.go [hunk 3]
```diff
@@ -138,7 +142,19 @@ func (hd *headerDownloader) downloadUsingWaypoints(ctx context.Context, waypoint
 			"peerCount", peerCount,
 		)
 
-		headerBatches := make([][]*types.Header, len(waypointsBatch))
+		//
```
Comment (taratorio) [inline polygon/sync/header_downloader.go:145]: already in the task list

**Review by battlmonstr — APPROVED** [review]

File: polygon/sync/sync.go [hunk 4]
```diff
@@ -150,39 +152,67 @@ func (s *Sync) onNewHeaderEvent(
...
-		err = s.onNewHeaderEvent(ctx, newHeaderEvent, ccBuilder)
+		err = s.onNewBlockEvent(ctx, newBlockEvent, ccBuilder)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func (s *Sync) fetchNewBlocks(
```
Comment (battlmonstr) [inline polygon/sync/sync.go:183]: could we move this into the p2p service?

Comment (taratorio) [reply polygon/sync/sync.go]: ✅

**Pull request merged**